Added "github" context to users created during github sync

begedin opened the pull request:
Closes #948 

`Issues.UserLinker` and `IssueComment.UserLinker` ended up needing the exact same change and are quite similar to begin with. Instead of trying to force this into a `Github.Event.Common` module, I figured this is a good opportunity to start a `CodeCorps.Account` bounded context, with its first function, `Account.create_from_github/1`

- added `Account` bounded context
- added `Account.create_from_github/1`
- added `AccountTest`
- added `fixtures/github/endpoint/user.json`

The context is simply `"github"` since the whole idea is that the user being created originates from github.

joshsmith commented:
The Phoenix docs on contexts seem to indicate that a pluralized form – `Accounts` – might make more sense here. How do you feel about this?

joshsmith commented on lib/code_corps/account/account.ex:
```diff
@@ -0,0 +1,26 @@
+    %User{}
+    |> Changeset.change(attrs |> Adapters.User.from_github_user())
+    |> Changeset.put_change(:context, "github")
+    |> Changeset.unique_constraint(:email)
```
What happens if the email is `nil`? GitHub is not guaranteed to return one.

begedin replied:
We already merged in a PR where a user can be inserted without an email, so we should be fine. Really, what's important for us here is the github id, so we can associate this "stub" with an actual user later, when they connect with github.

begedin commented:
> The Phoenix docs on contexts seem to indicate that a pluralized form – Accounts – might make more sense here. How do you feel about this?

I don't feel strongly about it one way or another. I can make the quick change

begedin committed:
Added "github" context to users created during github sync

- added Accounts bounded context
- added Accounts.create_from_github
- added AccountsTest
- added user.json github fixture
